components/Filter.coffee

```diff
@@ -1,46 +1,36 @@
 noflo = require("noflo")
 
-class Filter extends noflo.Component
-
-  description: "filters an IP which is a string using a regex"
-
-  constructor: ->
-    @regex = null
-
-    @inPorts = new noflo.InPorts
-      in:
-        datatype: 'string'
-        description: 'String to filter'
-      pattern:
-        datatype: 'string'
-        description: 'String representation of a regexp used as filter'
-    @outPorts = new noflo.OutPorts
-      out:
-        datatype: 'string'
-        description: 'String passing the filter'
-      missed:
-        datatype: 'string'
-        description: 'String failing the filter'
-
-    @inPorts.pattern.on "data", (data) =>
-      @regex = new RegExp(data)
-
-    @inPorts.in.on "begingroup", (group) =>
-      @outPorts.out.beginGroup(group)
-
-    @inPorts.in.on "data", (data) =>
-      unless typeof data is 'string'
-        data = (data).toString()
-      if @regex? and data?.match?(@regex)?
-        @outPorts.out.send data
-        return
-      @outPorts.missed.send data if @outPorts.missed.isAttached()
-
-    @inPorts.in.on "endgroup", (group) =>
-      @outPorts.out.endGroup()
-
-    @inPorts.in.on "disconnect", =>
-      @outPorts.out.disconnect()
-      @outPorts.missed.disconnect() if @outPorts.missed.isAttached()
-
-exports.getComponent = -> new Filter
+exports.getComponent = ->
+  c = new noflo.Component
+  c.description = "filters an IP which is a string using a regex"
+
+  c.inPorts.add 'in',
+    datatype: 'string'
+    description: 'String to filter'
+  c.inPorts.add 'pattern',
+    datatype: 'string'
+    description: 'String representation of a regexp used as filter'
+    control: true
+    required: true
+  c.outPorts.add 'out',
+    datatype: 'string'
+    description: 'String passing the filter'
+  c.outPorts.add 'missed',
```

Comment:
It could be reasonable to enable bracket forwarding to this port as well:

```coffeescript
forwardBrackets:
  in: ['out', 'missed']
```

but here and for any components splitting the input stream beware of empty brackets.